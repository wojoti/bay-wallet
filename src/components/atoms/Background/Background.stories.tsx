import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {Background} from 'components/atoms';
import {Text} from 'react-native';
import {BackgroundProps} from './Background';

export default {
  title: 'Atoms/Background',
  component: Background,
} as ComponentMeta<typeof Background>;
const Template: ComponentStory<typeof Background> = (args: BackgroundProps) => (
  <Background {...args}>
    <Text>Example text</Text>
    <Text>Example text</Text>
    <Text>Example text</Text>
  </Background>
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
  source: require('img/wave.png'),
};

export const ExampleWithCustomStyle = Template.bind({});
ExampleWithCustomStyle.args = {
  testId: '',
  source: require('img/wave.png'),
  customStyle: {padding: 20},
};
