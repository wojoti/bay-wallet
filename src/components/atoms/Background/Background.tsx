import {PropsWithChildren} from 'react';
import {
  ImageBackground,
  ImageSourcePropType,
  StyleProp,
  ViewStyle,
} from 'react-native';
import style from './Background.style';

export type BackgroundProps = PropsWithChildren<{
  source: ImageSourcePropType;
  testId?: string;
  customStyle?: StyleProp<ViewStyle>;
}>;

const Background = ({
  testId,
  source,
  customStyle,
  children,
}: BackgroundProps) => {
  return (
    <ImageBackground
      source={source}
      style={[style.wrapper, customStyle]}
      testID={testId}>
      {children}
    </ImageBackground>
  );
};

export default Background;
