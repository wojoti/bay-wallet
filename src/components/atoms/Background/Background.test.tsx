import '@testing-library/jest-native/extend-expect';
import {render, screen} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import Background, {BackgroundProps} from './Background';

const props: BackgroundProps = {
  testId: 'test-background-id',
  source: require('img/wave.png'),
};

const modifiedProps: BackgroundProps = {
  ...props,
  customStyle: {width: '100%'},
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<Background {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render Background', () => {
  render(<Background {...props} />);
  const BackgroundElement = screen.getByTestId('test-background-id');
  expect(BackgroundElement).toBeOnTheScreen();
});

test('should render background with modified style through props', () => {
  render(<Background {...modifiedProps} />);
  const backgroundElement = screen.getByTestId('test-background-id');
  expect(backgroundElement).toBeOnTheScreen();
  expect(backgroundElement).toHaveStyle({width: '100%'});
});
