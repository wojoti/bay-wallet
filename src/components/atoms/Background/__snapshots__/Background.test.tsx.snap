// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/img/wave.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
    testID="test-background-id"
  />
</View>
`;
